pkg/controller/cainjector/indexers.go

```diff
@@ -68,82 +89,52 @@ func buildCertToInjectableFunc(listTyp runtime.Object, resourceName string) cert
 	}
 }
 
-// secretForCertificateMapper is a Mapper that converts secrets up to injectables, through certificates.
-type secretForCertificateMapper struct {
-	Client                  client.Reader
-	log                     logr.Logger
-	certificateToInjectable certificateToInjectableFunc
-}
-
-func (m *secretForCertificateMapper) Map(obj client.Object) []ctrl.Request {
-	// grab the certificate, if it exists
-	certName := owningCertForSecret(obj.(*corev1.Secret))
-	if certName == nil {
-		return nil
-	}
-
-	secretName := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
-	log := m.log.WithValues("secret", secretName, "certificate", *certName)
-
-	var cert cmapi.Certificate
-	// confirm that a service owns this cert
-	if err := m.Client.Get(context.Background(), *certName, &cert); err != nil {
-		// TODO(directxman12): check for not found error?
-		log.Error(err, "unable to fetch certificate that owns the secret")
-		return nil
-	}
-
-	return m.certificateToInjectable(log, m.Client, *certName)
-}
-
-// certMapper is a mapper that converts Certificates up to injectables
-type certMapper struct {
-	Client       client.Reader
-	log          logr.Logger
-	toInjectable certificateToInjectableFunc
-}
-
-func (m *certMapper) Map(obj client.Object) []ctrl.Request {
-	certName := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
-	log := m.log.WithValues("certificate", certName)
-	return m.toInjectable(log, m.Client, certName)
-}
-
```

Comment:
All this functionality was removed and instead each of the 3 functions that get passed to the event handler come from a closure that gives it access to client and logger- would hopefully make it easier to follow how we get from the resource for which event handler got called (i.e secret) to the resource for which the reconcile loop is triggered (i.e validatingwebhookconfiguration)